compat/mingw.c

```diff
@@ -426,6 +428,47 @@ static wchar_t *normalize_ntpath(wchar_t *wbuf)
 	return wbuf;
 }
 
+/* rename and unlink a file. This is needed for the ability to recreate a file
+ * with the same name, for example on checkout which unlinks and recreates the file */
+static int rename_and_unlink(const wchar_t *pathname, int (*unlinker)(const wchar_t *))
+{
+	wchar_t temppath[MAX_LONG_PATH];
+	int tries = 0;
+	*temppath = 0;
+	/* leave some space for the .unlink_XXXXXX extension */
+	wcsncat(temppath, pathname, sizeof(temppath) - 15);
+	wcsncat(temppath, L".unlink_XXXXXX", sizeof(temppath));
```

Comment:
> But if we get a name that is too long, will MoveFile work?

No. https://github.com/git-for-windows/git/blob/master/compat/mingw.h#L531

Reply:
Then do we agree on this part?

Reply:
```
/**
 * Max length of long paths (exceeding MAX_PATH). The actual maximum supported
 * by NTFS is 32,767 (* sizeof(wchar_t)), but we choose an arbitrary smaller
 * value to limit required stack memory.
 */
#define MAX_LONG_PATH 4096
```

The constant is defined such that we can afford to add a few characters to the end.

Reply:
```
/**
 * Max length of long paths (exceeding MAX_PATH). The actual maximum supported
 * by NTFS is 32,767 (* sizeof(wchar_t)), but we choose an arbitrary smaller
 * value to limit required stack memory.
 */
#define MAX_LONG_PATH 4096
```

The constant is defined such that we can afford to add a few characters to the end.

Reply:
Done